Engine/ac/parser.cpp

```diff
@@ -269,7 +269,7 @@ int parse_sentence (const char *src_text, int *numwords, short*wordarray, short*
                 // if it's an unknown word, store it for use in messages like
                 // "you can't use the word 'xxx' in this game"
                 if ((word < 0) && (play.bad_parsed_word[0] == 0))
-                    strcpy(play.bad_parsed_word, thisword);
+                    strncpy(play.bad_parsed_word, thisword, 100);
```

Comment:
Please use snprintf instead of strncpy, strncpy *does not guarantee null-terminator*:
https://en.cppreference.com/w/c/string/byte/strncpy

Another variant is to use newer strncpy_s, it seems, but I've been replacing str(n)cpy to snprintf for quite some time, so that might be more consistent.

Reply:
I replaced it with `snprintf(play.bad_parsed_word, 100, "%s", thisword);`